### PR TITLE
ZCS-14938: Fix BHR SOAP failures

### PR DESCRIPTION
**Pull request by pallavikhairnar21**
**Fix following failing test cases in soap run:**

- data/soapvalidator/Auth/ForgetPassword/ZCS-4798.xml
- data/soapvalidator/Auth/ForgetPassword/ZCS-4800.xml
- data/soapvalidator/Calendar/Appointments/CreateAppointmentRequest-Reminder.xml
- data/soapvalidator/Tasks/Get-Tasks.xml

### CODE REVIEW

**Review by nilamraut24 — APPROVED** [review]
Looks good to me.

**Review by shrutig0510 — APPROVED** [review]

File: data/soapvalidator/Calendar/Appointments/CreateAppointmentRequest-Reminder.xml [hunk 1]
```diff
@@ -142,7 +142,7 @@
         <t:response>
             <t:select path="//mail:GetMsgResponse/mail:m[@id='${appt1.invId}']">
                         <t:select path="//mail:comp">
-                                <t:select path="//mail:s" attr="d" match="${ICALTIME[${time.20071201120000.gmt}]}"/>
+                                <t:select path="//mail:s" attr="d" contains="${ICALTIME[${time.20071201120000.gmt}]}"/>
```
Comment (prashantajari) [inline data/soapvalidator/Calendar/Appointments/CreateAppointmentRequest-Reminder.xml:145]: Do you know why Z was added in start and end time? This testcase is not creating appointment with timezone.

File: data/soapvalidator/Tasks/Get-Tasks.xml [hunk 2]
```diff
@@ -141,8 +141,8 @@
         <t:response>
             <t:select path = "//mail:GetTaskResponse/mail:task/mail:inv/mail:comp" attr="name" match="${task1.subject}"/>
             <t:select path = "//mail:GetTaskResponse/mail:task/mail:inv/mail:comp" attr="loc" match="${task1.location}"/>
-            <t:select path = "//mail:GetTaskResponse/mail:task/mail:inv/mail:comp/mail:s" attr="d" match="${Time1}"/>
-            <t:select path = "//mail:GetTaskResponse/mail:task/mail:inv/mail:comp/mail:e" attr="d" match="${Time2}"/>
+            <t:select path = "//mail:GetTaskResponse/mail:task/mail:inv/mail:comp/mail:s" attr="d" contains="${Time1}"/>
```
Comment (prashantajari) [inline data/soapvalidator/Tasks/Get-Tasks.xml:144]: Do you know why Z was added in start and end time? This testcase is not creating task with timezone.

Comment (pallavikhairnar21) [reply data/soapvalidator/Tasks/Get-Tasks.xml:144]: Z is shorthand for the UTC timezone.

Comment (prashantajari) [reply data/soapvalidator/Tasks/Get-Tasks.xml:144]: In above case, timezone info was not passed in create appointment/task request. We should keep match validation and validate for Z.

Comment (pallavikhairnar21) [reply data/soapvalidator/Tasks/Get-Tasks.xml]: Fixed test cases to add match validation and validation of Z

**Review by prashantajari — APPROVED** [review]

**Pull request merged**